tutorials/haskell/golden-testing-in-haskell-with-hspec-gold/tutorial.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Golden testing in Haskell with Hspec-Golden
+published: 2016-08-10
+tags: haskell, golden testing, snapshot testing, expect testing
+libraries: hspec-golden-1.0.0
+language: haskell
+author: Adrian Vera
+author-name: Adrian Vera
+twitter-profile: adrian_med
+github-profile: medAdrian
+description: A golden test is an IO action that writes its result to a file. To pass the test, this output file should be identical to the corresponding golden file, which contains the correct result for the test.
+---
+
+Golden tests are like unit tests, with the difference that the expected output is stored in a separate file. 
+
+Golden tests can also be known as:
+* Golden master testing
+* Approval testing
+* Snapshot testing
+* Expect testing
+
+There are many forms of getting the same result with different approaches, as the [Jest](https://jestjs.io/docs/en/snapshot-testing.html) snapshot testing,
+comparing two screenshots as [Wraith](https://github.com/BBC-News/wraith) or with the [Approval testing method](https://dzone.com/articles/approval-tests-alternative).
+
+##The Main idea
+A golden test is just an **IO action** that writes its result to a file.
+To pass the test, the output **file** should be identical to the corresponding golden file,
+that contains the correct result for the given test.
+
+####‘A Golden File is the absolute source of truth. You validate your response against the Golden File. In a nutshell, the Golden file contains the response you expect from your test’
+
+##Changes on test flow
+The workflow for running the tests now becomes the following:
+
+* Run a test and make sure it fails.
+* Review the response from your test to see if it is accurate.
+* Run go test -update to update the existing golden file (or create new ones).
+* Re-run go test to ensure the test does not fail now.
+
+##How do Golden tests differentiate from unit tests?
+* If the output is larger in size, it's not practical to put it in the source code.
+
+* There is no need to escape quotes or binary data in the output.
+
+* When you add a new test, your testing framework can generate the missing golden file from the current output of the function.
+
+* It is best if you can write down the expected output without looking at the actual output, but it is not always possible.
+
+* The expected output can be automatically updated.
+
+* You can tell your test framework to update all golden files from the current outputs, then check git diff to ensure that all changes are valid, and commit.
+
+* If some of your tests suddenly started failing, you can use diff or other such tools to compare the golden file to the actual file and figure out what exactly changed.
+
+##What is Hspec-Golden?
+Hspec-Golden is a superset for the popular Haskell B.D.D. testing framework Hspec,
```

Comment:
Perhaps you should include the reference for the library's homepage: http://hackage.haskell.org/package/hspec-golden or GH repo.

Reply:
I forgot to, thanks for the reminder!, i will add this on the next commit.